process/dataValidators/txValidator.go

```diff
@@ -77,9 +84,11 @@ func (txv *txValidator) CheckTxValidity(interceptedTx process.InterceptedTransac
 	}
 
 	// for relayed v3, we allow sender accounts that do not exist
+	// only if the transaction is not guarded
 	isRelayedV3 := common.IsRelayedTxV3(interceptedTx.Transaction())
 	hasValue := hasTxValue(interceptedTx)
-	shouldAllowMissingSenderAccount := isRelayedV3 && !hasValue
+	isGuardedTx := txv.isGuardedTxAfterSupernova(interceptedTx.Transaction())
```

Comment:
This check is somehow non-trivial, and now it will get executed for every transaction.

Maybe nitpicking, but - would there be an overhead?